### PR TITLE
Remove `DelegateInvokeStubManager`

### LINKED ISSUE #85339: Native->Managed (.net core) step-in not working in Visual Studio
### Description

When interop debugging an app running .net core and stopped at a native frame, attempting to step into a managed function call fails to go into the managed side. 

Upon some investigation by us (the vs debugger), it appears that the same scenario in .net framework will send us a step complete event when this step in is requested. We then are able to correctly stop at the right spot on the managed side. In .net core, this event never comes in so we just step over the call to the managed function. We have our stepper use ICorDebugStepper->Step(stepIn:true)

I am unsure if there's some callback we need to provide or some additional setup so that we get notified about the step completion. There is a good chance this is a long-standing issue. Recent efforts to increase our net core interop testing revealed this issue.

This sample app shows the issue happening (open the .sln file)
[ConsoleApp1.zip](https://github.com/dotnet/runtime/files/11325022/ConsoleApp1.zip)

### Reproduction Steps

- Start in a native frame
- Step to a call to a managed function
- Step-in

### Expected behavior

Step into managed side works

### Actual behavior

We step over the managed function call 

### Regression?

_No response_

### Known Workarounds

Enabling JMC partially hides this behavior. If the managed function being stepped into is user code, the step in works as expected.

### Configuration

.net 6 and .net 8 preview both show this behavior

### Other information

_No response_

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
All Delegate Invoke scenarios now use the `StubLinker`. This means there is an inherent confusion between the two stub managers. Upon further inspection combining these two by adding a new flag on the `Stub` class means they can be combined logically and simplify the code path by removing a `StubManager`.

Fixes #85339

### CODE REVIEW

**Review by hoyosjs — APPROVED** [review]

**Review by noahfalk — APPROVED** [review]
👍

**Comment by AaronRobinsonMSFT** [comment]
The failures are known. I reran the tests before checking for known issues.

File: src/coreclr/vm/stublink.h [hunk 1]
```diff
@@ -464,6 +465,7 @@ class Stub
         LOADER_HEAP_BIT         = 0x20000000,
         INSTANTIATING_STUB_BIT  = 0x10000000,
         UNWIND_INFO_BIT         = 0x08000000,
+        THUNK_BIT               = 0x04000000,
 
         CODEBYTES_MASK          = UNWIND_INFO_BIT - 1,
```
Comment (AustinWise) [inline src/coreclr/vm/stublink.h:470]: Should `CODEBYTES_MASK` be updated to account for the new flag?

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/stublink.h:470]: Ugh. Yes. Thanks for catching this. Thankfully it doesn't matter at present, but it needs to be updated.

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/stublink.h:470]: https://github.com/dotnet/runtime/pull/88830

**Review by hoyosjs — APPROVED** [review]

**Pull request merged**